src/main/kotlin/io/perfometer/runner/CoroutinesScenarioRunner.kt

```diff
@@ -1,20 +1,25 @@
 package io.perfometer.runner
 
 import io.perfometer.dsl.HttpStep
+import io.perfometer.dsl.ParallelStep
 import io.perfometer.dsl.PauseStep
 import io.perfometer.dsl.RequestStep
 import io.perfometer.http.client.HttpClient
 import io.perfometer.statistics.PauseStatistics
 import io.perfometer.statistics.ScenarioSummary
 import kotlinx.coroutines.*
 import java.time.Duration
+import java.util.concurrent.ConcurrentLinkedDeque
 import kotlin.time.ExperimentalTime
 import kotlin.time.toKotlinDuration
 
 internal class CoroutinesScenarioRunner(
     httpClient: HttpClient,
 ) : BaseScenarioRunner(httpClient) {
 
+    // todo @ttarczynski - consider moving this to a separate class
+    private val parallelJobs = ConcurrentLinkedDeque<Job>()
```

Comment:
Queue of parallel jobs should be separate for every coroutine (like HttpClient -- see in cookies support implementation).